connector_magento/tests/common.py

```diff
@@ -0,0 +1,313 @@
+        return self.code
+
+
+@contextmanager
+def mock_urlopen_image():
+    with mock.patch("requests.get") as requests_get:
+        requests_get.return_value = MockResponseImage("")
+        yield
+
+
+class MagentoHelper(object):
+    def __init__(self, cr, registry, model_name):
+        self.cr = cr
+        self.model = registry(model_name)
+
+    def get_next_id(self):
+        self.cr.execute("SELECT max(external_id::int) FROM %s " % self.model._table)
+        result = self.cr.fetchone()
+        if result:
+            return int(result[0] or 0) + 1
+        else:
+            return 1
+
+
+class MagentoTestCase(SavepointComponentCase):
+    """ Base class - Test the imports from a Magento Mock.
+
+    The data returned by Magento are those created for the
+    demo version of Magento on a standard 1.9 version.
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.recorder = recorder
+        # disable commits when run from pytest/nosetest
+        odoo.tools.config["test_enable"] = True
+
+        self.backend_model = self.env["magento.backend"]
+        warehouse = self.env.ref("stock.warehouse0")
+        self.backend = self.backend_model.create(
+            {
+                "name": "Test Magento",
+                "version": "1.7",
+                "location": "http://magento",
+                "username": "odoo",
+                "warehouse_id": warehouse.id,
+                "password": "odoo42",
+            }
+        )
+        # payment method needed to import a sale order
+        self.workflow = self.env.ref("sale_automatic_workflow.manual_validation")
+        check_journal = self.env["account.journal"].search(
+            [("type", "=", "cash"), ("code", "=", "Check")], limit=1
+        )
+        if not check_journal:
+            check_journal = self.env["account.journal"].create(
+                {"name": "Check", "type": "cash", "code": "Check"}
+            )
+        self.journal = check_journal
+
+        payment_method = self.env.ref("account.account_payment_method_manual_in")
+        for name in ["checkmo", "ccsave", "cashondelivery"]:
+            self.env["account.payment.mode"].create(
+                {
+                    "name": name,
+                    "workflow_process_id": self.workflow.id,
+                    "import_rule": "always",
+                    "days_before_cancel": 0,
+                    "bank_account_link": "fixed",
+                    "payment_method_id": payment_method.id,
+                    "fixed_journal_id": self.journal.id,
+                }
+            )
+        for tax_percent in [8.25, 9.0]:
+            account_tax_form = Form(self.env["account.tax"])
+            account_tax_form.name = "Sale {}{}".format(tax_percent, "%")
+            account_tax_form.amount = tax_percent
+            account_tax_form.save()
+
+        journal_bank = self.env["account.journal"].search(
+            [("type", "=", "bank")], limit=1
+        )
```

Comment:
Shouldn't you check that this actually returns what is needed and create one otherwise, like what is done above for the Cash Journal?